homeassistant/components/habitica/entity.py

```diff
@@ -29,26 +31,88 @@ def __init__(
         self,
         coordinator: HabiticaDataUpdateCoordinator,
         entity_description: EntityDescription,
+        subentry: ConfigSubentry | None = None,
     ) -> None:
         """Initialize a Habitica entity."""
         super().__init__(coordinator)
         if TYPE_CHECKING:
             assert coordinator.config_entry.unique_id
         self.entity_description = entity_description
-        self._attr_unique_id = (
-            f"{coordinator.config_entry.unique_id}_{entity_description.key}"
+        self.subentry = subentry
+        unique_id = (
+            subentry.unique_id
+            if subentry is not None and subentry.unique_id
+            else coordinator.config_entry.unique_id
         )
+
+        self._attr_unique_id = f"{unique_id}_{entity_description.key}"
         self._attr_device_info = DeviceInfo(
             entry_type=DeviceEntryType.SERVICE,
             manufacturer=MANUFACTURER,
             model=NAME,
-            name=coordinator.data.user.profile.name,
+            name=self.user.profile.name,
             configuration_url=(
-                URL(coordinator.config_entry.data[CONF_URL])
-                / "profile"
-                / coordinator.config_entry.unique_id
+                URL(coordinator.config_entry.data[CONF_URL]) / "profile" / unique_id
             ),
-            identifiers={(DOMAIN, coordinator.config_entry.unique_id)},
+            identifiers={(DOMAIN, unique_id)},
+        )
+
+        if subentry:
+            self._attr_device_info.update(
+                DeviceInfo(
+                    via_device=(
+                        (
+                            DOMAIN,
+                            f"{coordinator.config_entry.unique_id}_{self.user.party.id!s}",
+                        )
+                    )
+                )
+            )
+
+    @property
+    def user(self) -> UserData:
+        """Return the user data."""
+        return self.coordinator.data.user
+
+
+class HabiticaPartyMemberBase(HabiticaBase):
+    """Base Habitica party member entity."""
+
+    def __init__(
+        self,
+        coordinator: HabiticaDataUpdateCoordinator,
+        party_coordinator: HabiticaPartyCoordinator,
+        entity_description: EntityDescription,
+        subentry: ConfigSubentry | None = None,
+    ) -> None:
+        """Initialize a Habitica entity."""
+        self.party_coordinator = party_coordinator
+        super().__init__(coordinator, entity_description, subentry)
+
+    @property
+    def user(self) -> UserData:
+        """Return the user data of the party member."""
+        if TYPE_CHECKING:
+            assert self.subentry
+            assert self.subentry.unique_id
+        return self.party_coordinator.data.members[UUID(self.subentry.unique_id)]
```

Comment:
instead of adding check for `available` everywhere this is called, I think it would be safer to just make this method return `UserData | None`. That way the methods using this just check for `None` which is much clearer than why they need to check `available`.